tdp/core/collections.py

```diff
@@ -239,3 +252,24 @@ def check_operations_exist(self, operations_names: List[str]) -> None:
                 raise MissingOperationError(
                     f"Operation {operation_name} not found in collections."
                 )
+
+    def check_operations_hosts_exist(
+        self, operation_names: Iterable[str], host_names: Iterable[str]
+    ) -> None:
+        """Check that all operations exist and hosts exist for all operations.
+
+        Args:
+            operation_names: Iterable of operation names to check.
+            host_names: Iterable of host names to check for each operation.
+
+        Raises:
+            MissingHostForOperationError: If a host name is missing for an operation.
+        """
+        self.check_operations_exist(operation_names)
+        for operation_name in operation_names:
+            operation = self.operations[operation_name]
```

Comment:
You could use the `self.getOperation` which allow to remove `self.check_operations_exist(operation_names)`.

Reply:
Done.